### DISCUSSION TITLE
v3.11 to v4.0.0-rc.4 msi that installs certificates fails

### DISCUSSION THREAD

**Discussion by wmanning** [wixtoolset/issues#7359] [Questions] [answered]
Question for the WiX team.  I converted all of my MSI/burn packages from WiX 3.11 to v4.0.0-rc.4.

I have a msi where that installs certificates using the Iis extension Certificate element.
It fails to install.  I had ```xml <InstallerPlatform>x86</InstallerPlatform>``` so I was thinking since it's a 64bit machine maybe I should try ```xml <InstallerPlatform>x64</InstallerPlatform>```.  No difference: same error.

Here is partial MSI log showing the error:  (note the Wix4InstallCertificates_X64 in the log, this is when I changed InstallerPlatform to x64)

> MSI (s) (90:E0) [17:20:24:768]: Doing action: Wix4InstallCertificates_X64
MSI (s) (90:E0) [17:20:24:768]: Note: 1: 2205 2:  3: ActionText 
Action ended 17:20:24: InstallFiles. Return value 1.
MSI (s) (90:B0) [17:20:24:775]: Invoking remote custom action. DLL: C:\Windows\Installer\MSIF82F.tmp, Entrypoint: InstallCertificates
MSI (s) (90!80) [17:20:24:801]: Note: 1: 2205 2:  3: CertificateHash 
MSI (s) (90!80) [17:20:24:801]: Note: 1: 2228 2:  3: CertificateHash 4: SELECT * FROM `CertificateHash` 
Action start 17:20:24: Wix4InstallCertificates_X64.
InstallCertificates:  Error 0x8007064f: failed to open view on database
InstallCertificates:  Error 0x8007064f: failed to openexecute temp view with query SELECT * FROM `CertificateHash`
InstallCertificates:  Error 0x8007064f: Failed to add encoded has for certificate: CERT_DynaCert1518
InstallCertificates:  Error 0x8007064f: Failed to resolve certificate: CERT_DynaCert1518
CustomAction Wix4InstallCertificates_X64 returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)

The msi project consists of three wxs files; a bit large to post here.

Could this be a bug in Wix 4?  

I can try make a simple repo if you feel it could be a bug.
I will troubleshoot further if not.

**Comment by barnson** [comment, marked as answer]
It's a bug so feel free to open an issue, but I think I have it narrowed down.

**Reply by wmanning** [reply]
Bug opened: https://github.com/wixtoolset/issues/issues/7360
